client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,171 @@
+			<h1 className="marketplace-domain-upsell__header marketplace-title">Choose a domain</h1>
+			<h2 className="marketplace-domain-upsell__header marketplace-subtitle">
+				Yoast SEO requires a top level domain to index your site and help you rank higher.
+			</h2>
+		</div>
+	);
+}
+
+function DomainPickerContainer( { onDomainSelect, selectedDomain } ) {
+	return (
+		<DomainPicker
+			header={ <Header /> }
+			analyticsUiAlgo={ ANALYTICS_UI_LOCATON_MARKETPLACE_DOMAIN_SELECTION }
+			analyticsFlowId={ MARKETPLACE_FLOW_ID }
+			onDomainSelect={ onDomainSelect }
+			currentDomain={ selectedDomain }
+			hideRecommendationLabel={ true }
+		/>
+	);
+}
+
+function MarketplaceShoppingCart( { onAddDomainToCart, selectedDomain } ) {
+	const { responseCart, isLoading } = useShoppingCart();
+	const { products, sub_total_display } = responseCart;
+
+	return isLoading ? (
+		<div></div>
+	) : (
+		<>
+			<h1 className="marketplace-domain-upsell__shopping-cart marketplace-title">Your cart</h1>
+			<div className="marketplace-domain-upsell__shopping-cart basket-items">
+				{ products.map( ( product ) => {
+					return (
+						<div
+							key={ product.product_name }
+							className="marketplace-domain-upsell__shopping-cart basket-item"
+						>
+							<div>{ product.product_name }</div>
+							<div>{ product.item_subtotal_display }</div>
+						</div>
+					);
+				} ) }
+			</div>
+			<hr />
+			<h1 className="marketplace-domain-upsell__shopping-cart total">
+				<div> Total </div>
+				<div>{ sub_total_display }</div>
+			</h1>
+
+			<Button
+				onClick={ onAddDomainToCart }
+				isBusy={ false }
+				isPrimary
+				disabled={ selectedDomain === null }
+			>
+				Checkout
+			</Button>
+		</>
+	);
+}
+
+function CalypsoWrappedMarketplaceDomainUpsell(): JSX.Element {
+	const [ selectedDomain, setDomain ] = useState( null );
+	const { addProductsToCart, replaceProductsInCart } = useShoppingCart();
+	const products = useSelector( getProductsList );
+	const isFetchingProducts = useSelector( isProductsListFetching );
+	const selectedSite = useSelector( getSelectedSite );
+
+	useEffect( () => {
+		//FIXME: This code segment simulates yoast premium already being added when arriving here. To be removed when plugins page is completed.
+		if ( ! isFetchingProducts && products[ 'yoast_premium' ] ) {
+			const yoastProduct = fillInSingleCartItemAttributes(
+				{ product_slug: 'yoast_premium' },
+				products
+			);
+			replaceProductsInCart( [ { ...yoastProduct } ] );
+		}
+	}, [ isFetchingProducts, products, replaceProductsInCart ] );
+
+	const onAddDomainToCart = () => {
+		const { product_slug, domain_name } = selectedDomain;
+		const domainProduct = {
+			...domainRegistration( {
+				productSlug: product_slug,
+				domain: domain_name,
+				source: 'Marketplace-Yoast-Domain-Upsell',
+			} ),
+			...selectedDomain,
+		};
+		addProductsToCart( [ domainProduct ] );
+		window.location.href = window.location.origin + '/checkout/' + selectedSite.slug;
```

Comment:
A full page reload should not be needed here. And that's what `window.location` does. Let's do a normal SPA navigation with `page()` here.

Also, other places that `addProductsToCart` wait until the adding is done before redirecting:
```js
addProductsToCart( [ domainProduct ] ).then( () => {
  page( '/checkout/' + selectedSite.slug );
} );
```

Reply:
Fixed by 0aa0786